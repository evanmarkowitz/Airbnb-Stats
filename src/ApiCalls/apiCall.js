

export const fetchApartments = async (apiKey, neighborhood, roomType) => {
  try {
    const options = {
      method: 'GET',
    }
    const response = await fetch(`https://public.opendatasoft.com/api/records/1.0/search/?dataset=air-bnb-listings&rows=50&key=${apiKey}&facet=neighbourhood&facet=room_type&facet=column_10&facet=calculated_host_listings_count&facet=city&facet=column_19&refine.city=New-york-city&refine.neighbourhood=${neighborhood}&refine.room_type=${roomType}&exclude.number_of_reviews=0` , options)
    if (!response.ok) {
      throw new Error('error')
    }
    const data = await response.json()
    return data 
  }
  catch (error) {
    throw new Error(error.message)
  }
}

export const apartmentCleaner = (apts) => {
  let cleanedApts = apts.map(apt => {
    return {
      reviewsPerMonth: apt.fields.reviews_per_month,
      id: apt.fields.id,
      numReviews: apt.fields.number_of_reviews,
      price: apt.fields.column_10,
      roomType: apt.fields.room_type,
      name: apt.fields.name
    }
  }) 
  return cleanedApts
}
