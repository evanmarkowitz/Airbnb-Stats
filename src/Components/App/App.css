.app {

  background-size: cover;
  height: 100vh;
  font-family: 'Montserrat', sans-serif;
  font-weight: 200;
  box-sizing: border-box;
  padding: 15px;
}

.headline {
  margin: 0;
}

.statue {
  width: 50px;
  margin-right: 25px;
}

.header{
  display: flex;
  justify-content: space-between;
}

