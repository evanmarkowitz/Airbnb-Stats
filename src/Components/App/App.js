import React, { Component } from 'react';
import './App.css';
import Filter from '../../Containers/Filter/Filter.js'
import Results from '../../Containers/Results/Results.js'
import { connect } from 'react-redux';
import statue from '../../Images/statue-of-liberty.svg'
import { fetchApartments, apartmentCleaner } from '../../ApiCalls/apiCall'
import {getApts, getCurrApt} from '../../actions/index.js'
import Overview  from '../../Containers/Overview/Overview'
import CurrApt from '../../Containers/CurrApt/CurrApt'



export class App extends Component {
  constructor() {
    super()
    this.state ={
      error: ''
    }
  }
  
 
  async componentDidMount() {
    let data;
    try {
      data = await fetchApartments()
    } catch(error) {
      this.setState({ error: error.message})
    }
    
    let cleanApartments = apartmentCleaner(data.records)
    await this.props.getApts(cleanApartments)
  }

  render() {
    return (
      <main className='app'>
      <header className='header'>
        <h1 className='headline'>AIRBNB PRICES BY NEIGHBORHOOD</h1>
        {this.state.error && <p>this.state.error</p>}
        <img className='statue' src={statue} alt='statue of liberty icon'></img>
      </header>
        <Filter />
        <section className= 'body'>
          <aside>
            <Overview />
            <CurrApt/>
          </aside>
          <Results apts={this.props.apts} hood={this.props.hood} 
          aptType={this.props.aptType} setCurrApt={this.props.getCurrApt} />
        </section>
      </main>
    );
  }
}

export const mapStateToProps = (state) => ({
  apts: state.apts,
  hood: state.hood,
  aptType: state.aptType
})


export const mapDispatchToProps = (dispatch) => ({
  getApts: (apts) => dispatch(getApts(apts)),
  getCurrApt: (apt) => dispatch(getCurrApt(apt)),
})

export default connect(mapStateToProps, mapDispatchToProps)(App)