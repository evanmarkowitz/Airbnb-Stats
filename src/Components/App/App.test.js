import {mapStateToProps, mapDispatchToProps, App} from './App'
import { getApts, getCurrApt } from '../../actions/index.js'
import {shallow} from 'enzyme'
import { fetchApartments, apartmentCleaner } from '../../ApiCalls/apiCall'
import React from 'react'



describe('App: Map Dispatch and StateTo Props', () => {
  let initialState;
  let mockDispatch
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />)
    initialState = {apts: [{name:'hi'}], hood: 'Williamsburg', aptType: 'Private Room'}
    mockDispatch = jest.fn();
    
  })
  it('should map snapshot', () =>{
    expect(wrapper).toMatchSnapshot()
  })

  it('should map getApt to props', () => {
    const mockAction = getApts(initialState.apts)
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getApts(initialState.apts);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })

  it('should map getCurrApt to props', () => {
    const mockAction = getCurrApt(initialState.currApt)
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getCurrApt(initialState.currApt);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })

  it('Should map state to props', () => {
    let props= initialState
    const mappedProps = mapStateToProps(props);
    expect(mappedProps).toEqual(props);
  })

  it.skip('should call fetch on component did mount', () => {
    let mockEvent = { preventDefault: jest.fn() }
    apartmentCleaner = jest.fn()
    window.fetch = jest.fn()
    wrapper.instance().componentDidMount()
    expect(apartmentCleaner).toHaveBeenCalled()
  })
})
