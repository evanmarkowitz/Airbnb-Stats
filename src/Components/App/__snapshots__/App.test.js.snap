// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Map Dispatch and StateTo Props should map snapshot 1`] = `ShallowWrapper {}`;

exports[`App: Map Dispatch and StateTo Props should map snapshot 2`] = `ShallowWrapper {}`;
